### PR TITLE
Implementing loss scaling scheduler callback and schedulers

### PR DESCRIPTION
**Pull request by laserkelvin**
This PR adds a new callback, called `LossScalingScheduler`, and scheduler classes that will modify the relative loss weights over the course of training.

- Two types of schedulers are implemented: `LinearScalingSchedule` and `SigmoidScalingSchedule`. The former will generate a linear ramp from start to end over steps or epochs, and the latter gives a gradual ramp up in the form of a sigmoid curve.
- The `LossScalingScheduler` is configured by passing schedules that are mapped to task keys, and for every training step or epoch (set by the schedules), applies a new value of the weighting to the appropriate task.
- The example script `examples/callbacks/loss_scheduling.py` is provided to show how it is configured.

This is useful for implementing curricula, where we prioritize learning of different properties over time.

### CODE REVIEW

**Comment by laserkelvin** [comment]
I think the same tests fail as in #266

File: matsciml/lightning/callbacks.py [hunk 1]
```diff
@@ -1492,3 +1511,95 @@ def on_fit_end(
...
+                try:
+                    new_scaling_value = schedule.step()
+                    pl_module.task_loss_scaling[target_key] = new_scaling_value
+                    self._logger.debug(
+                        f"Advanced {target_key} to new value: {new_scaling_value}"
+                    )
+                except StopIteration:
+                    self._logger.warning(
+                        f"{target_key} has run out of scheduled values; this may be unintentional."
+                    )
```
Comment (melo-gonzo) [inline matsciml/lightning/callbacks.py:1585]: Looks like you could combine this common bit between the on_x_end functions, and pass in a 'step' or 'epoch' string variable to use in the if statement and log message.

**Review by melo-gonzo — APPROVED** [review]
Just one general comment, otherwise looks good!

File: matsciml/lightning/callbacks.py (hunk 1 above)
Comment (laserkelvin) [reply matsciml/lightning/callbacks.py]: Done in d3b97da
